lib/node_modules/@stdlib/complex/parse-float64/test/test.js

```diff
@@ -0,0 +1,193 @@
+	var z;
+	var w;
+
+	z = new Complex128( 3.0, 0.0 );
+	w = parseComplex128( '3' );
+
+	t.strictEqual( isComplex128(w), true, 'is an instance' );
+	t.strictEqual( w.re, z.re, true, 'has expected property value' );
+	t.strictEqual( w.im, z.im, true, 'has expected property value' );
+
+	t.end();
+});
+
+tape( 'the function will parse a string representation of a complex number (only imaginary part)', function test( t ) {
+	var z;
+	var w;
+
+	z = new Complex128( 0.0, 8.5 );
+	w = parseComplex128( '8.5j' );
+
+	t.strictEqual( isComplex128(w), true, 'is an instance' );
+	t.strictEqual( w.re, z.re, true, 'has expected property value' );
+	t.strictEqual( w.im, z.im, true, 'has expected property value' );
+
+	t.end();
+});
+
+tape( 'the function will parse a string representation of a complex number (only real part, negative)', function test( t ) {
+	var z;
+	var w;
+
+	z = new Complex128( -3.753, 0.0 );
+	w = parseComplex128( '-3.753' );
+
+	t.strictEqual( isComplex128(w), true, 'is an instance' );
+	t.strictEqual( w.re, z.re, 'has expected real part' );
+	t.strictEqual( w.im, z.im, 'has expected imaginary part' );
+
+	t.end();
+});
+
+tape( 'the function will parse a string representation of a complex number (only imaginary part, negative)', function test( t ) {
+	var z;
+	var w;
+
+	z = new Complex128( 0.0, -0.3 );
+	w = parseComplex128( '-0.3j' );
+
+	t.strictEqual( isComplex128(w), true, 'is an instance' );
+	t.strictEqual( w.re, z.re, 'has expected real part' );
+	t.strictEqual( w.im, z.im, 'has expected imaginary part' );
+
+	t.end();
+});
+
+tape( 'the function correctly parses a string representation of a complex number with no space (a+ib format)', function test( t ) {
+	var z;
+	var w;
+
+	z = new Complex128( 5.0, 3.0 );
+	w = parseComplex128( '5+3j' );
+
+	t.strictEqual( isComplex128(w), true, 'is an instance' );
+	t.strictEqual( w.re, z.re, 'has expected real part' );
+	t.strictEqual( w.im, z.im, 'has expected imaginary part' );
+
+	t.end();
+});
+
+tape( 'the function correctly parses a string representation of a complex number with unconventional spacing (a+ ib format)', function test( t ) {
+	var z;
+	var w;
+
+	z = new Complex128( 5.0, 3.0 );
+	w = parseComplex128( '5+ 3j' );
+
+	t.strictEqual( isComplex128(w), true, 'is an instance' );
+	t.strictEqual( w.re, z.re, 'has expected real part' );
+	t.strictEqual( w.im, z.im, 'has expected imaginary part' );
+
+	t.end();
+});
+
+tape( 'the function will parse a string representation of a complex number with NaNs', function test( t ) {
+	var w = parseComplex128( 'NaN + NaNj' );
+
+	t.strictEqual( isComplex128(w), true, 'is an instance' );
+	t.ok( isNaN(w.re), 'has expected property value' );
+	t.ok( isNaN(w.im), 'has expected property value' );
+
+	t.end();
+});
+
+tape( 'the function will parse a string representation of a complex number with Infinity', function test( t ) {
+	var z;
+	var w;
+
+	z = new Complex128( Infinity, -Infinity );
```

Comment:
Do not use globals. Prefer `@stdlib/constants/float64/pinf` and `@stdlib/constants/float64/ninf`.